src/components/cesium_map/cesium_UI.js

```diff
@@ -130,20 +147,22 @@ class CesiumMap extends React.Component {
     // viewer.addPointsBySource(642092);
     viewer.visit(moorea);
     viewer.addHud("cesiumContainer");
+    // viewer.addLoading();
     viewer.trackMouseCoordinates(showCoordinates);
     viewer.enableTracking(selectedBoxCallbox);
     setPrimitive = new PointStreamPrimitiveCollection("Primitive Points");
     viewer.addPointPrimitives(setPrimitive);
     viewer.addDataSource(new PointStreamDatasource("BB points")).then((res) => { setPoints = res });
     searchFields = this.props.searchFields;
     updatePrimitive(lat, long)
-
     // set time interval to check the current view every 3 seconds and update points
     setInterval(() => {
-      if (Math.abs(viewer.currentView.latitude - lat) > 5 || Math.abs(viewer.currentView.longitude - long) > 5) {
+      let diffDistance = distance(lat, long, viewer.currentView.latitude, viewer.currentView.longitude);
+      // update the points every 5 seconds if two points differ in 50km + scale of height.
+      if (diffDistance > 50 + 4000 * viewer.currentView.height / 15000000) {
```

Comment:
This is still an arbitrary method. 50 kilometers is the minimum distance change to rerender the points and then add scaled distance based on the camera height.

Reply:
Even if it's arbitrary, it's worth explaining how you settled on the value.